stl/src/vector_algorithms.cpp

```diff
@@ -1496,5 +1497,114 @@ __declspec(noalias) size_t
     return __std_count_trivial<_Find_traits_8>(_First, _Last, _Val);
 }
 
+} // extern "C"
+
+#ifndef _M_ARM64EC
+namespace {
+    __m128i __forceinline _Bitset_to_string_1_step(const uint16_t _Val, const __m128i _Xor_v, const __m128i _And_v) {
+        const __m128i _Vec1  = _mm_cvtsi32_si128(_Val);
+        const __m128i _Vec2  = _mm_unpacklo_epi8(_Vec1, _Vec1);
+        const __m128i _Vec3  = _mm_unpacklo_epi8(_Vec2, _Vec2);
+        const __m128i _Vec4  = _mm_shuffle_epi32(_Vec3, _MM_SHUFFLE(0, 0, 1, 1));
+        const __m128i _Msk   = _mm_and_si128(_Vec4, _mm_set1_epi64x(0x0102040810204080));
+        const __m128i _Zeros = _mm_cmpeq_epi8(_Msk, _mm_setzero_si128());
+        const __m128i _Elems = _mm_xor_si128(_mm_and_si128(_Zeros, _Xor_v), _And_v);
+        return _Elems;
+    }
+
+    __m128i __forceinline _Bitset_to_string_2_step(const uint8_t _Val, const __m128i _Xor_v, const __m128i _And_v) {
+        const __m128i _Vec   = _mm_set1_epi16(_Val);
+        const __m128i _Msk   = _mm_and_si128(_Vec, _mm_set_epi64x(0x0001000200040008, 0x0010002000400080));
+        const __m128i _Zeros = _mm_cmpeq_epi16(_Msk, _mm_setzero_si128());
+        const __m128i _Elems = _mm_xor_si128(_mm_and_si128(_Zeros, _Xor_v), _And_v);
+        return _Elems;
+    }
+} // unnamed namespace
+#endif // !defined(_M_ARM64EC)
+
+extern "C" {
+
+__declspec(noalias) void __stdcall __std_bitset_to_string_1(
+    char* const _Dest, const void* _Src, size_t _Size_bits, const char _Elem0, const char _Elem1) noexcept {
+#ifndef _M_ARM64EC
+    if (_Use_sse2()) {
+        const __m128i _Xor_v = _mm_set1_epi8(_Elem0 ^ _Elem1);
+        const __m128i _And_v = _mm_set1_epi8(_Elem1);
+        if (_Size_bits > 16) {
+            char* _Pos = _Dest + _Size_bits;
+            _Size_bits &= 0xF;
+            char* const _Stop_at = _Dest + _Size_bits;
+            do {
+                uint16_t _Val;
+                memcpy(&_Val, _Src, 2);

```

Comment:
this `memcpy` is to avoid aliasing problems. But other places of this file do `reinterpret_cast` instead with a comment `NOTE: UNALIGNED MEMORY ACCESSES` and not even mentioning incorrect aliasing. Should this be done here too?

Reply:
I'm ok with the strictly correct `memcpy` although I suppose there's a consistency argument.